### PR TITLE
sql: correct type check error message on boolean exprs

### PR DESCRIPTION
**Pull request by madelynnblue**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3369)

### CODE REVIEW

File: sql/parser/type_check.go [hunk 1]
```diff
@@ -517,7 +517,7 @@ func typeCheckBooleanExprs(exprs ...Expr) (Datum, error) {
 			continue
 		}
 		if _, ok := dummyExpr.(DBool); !ok {
-			return nil, fmt.Errorf("incompatible AND argument type %s", dummyExpr.Type())
+			return nil, fmt.Errorf("incompatible %s argument type %s", op, dummyExpr.Type())
```
Comment (petermattis) [inline sql/parser/type_check.go]: Perhaps `s/type %s/type: %s/g`.

**Comment by petermattis** [comment]
LGTM

**Comment by mjibson** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/parser/type_check.go, line 520 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3369#-K576QGDZWhpJu193b6s-r1-520:-K577qldnNYVwPncVxhc:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0b781441474bdf43aa6e5ef4434f038568539aac/sql/parser/type_check.go#L520)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3369)*

**Comment by tamird** [comment]
LGTM

**Pull request merged**